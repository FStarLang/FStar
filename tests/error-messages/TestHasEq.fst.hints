[
  "\u0005¡cEmy£?Ñc³à\u0017B·d",
  [
    [
      "TestHasEq.mlist",
      1,
      2,
      1,
      [ "@query" ],
      0,
      "41956262b2868144aaaf8f8eaa9059cb"
    ],
    [
      "TestHasEq.__proj__MCons__item__hd",
      1,
      2,
      1,
      [
        "@MaxIFuel_assumption", "@query",
        "refinement_interpretation_Tm_refine_04c57beb84c7e65ffbf87f2b1dfc7ef5"
      ],
      0,
      "df32f280115ea9c1499fe739097d356c"
    ],
    [
      "TestHasEq.__proj__MCons__item__tl",
      1,
      2,
      1,
      [
        "@MaxIFuel_assumption", "@query",
        "refinement_interpretation_Tm_refine_04c57beb84c7e65ffbf87f2b1dfc7ef5"
      ],
      0,
      "e983288be308d61ddd93cee18bb9643b"
    ],
    [
      "TestHasEq.__proj__NCons__item__hd",
      1,
      2,
      1,
      [
        "@MaxIFuel_assumption", "@query",
        "refinement_interpretation_Tm_refine_7b907b210749d62a64a384804262b9e4"
      ],
      0,
      "3c0695628f70e00709305621686779a1"
    ],
    [
      "TestHasEq.__proj__NCons__item__tl",
      1,
      2,
      1,
      [
        "@MaxIFuel_assumption", "@query",
        "refinement_interpretation_Tm_refine_7b907b210749d62a64a384804262b9e4"
      ],
      0,
      "7a8bdde9a6655d284834ba5ae953e5a0"
    ],
    [
      "TestHasEq.test1",
      1,
      2,
      1,
      [
        "@MaxIFuel_assumption", "@query",
        "assumption_TestHasEq.mlist__uu___haseq",
        "assumption_TestHasEq.nlist__uu___haseq", "equation_Prims.eqtype",
        "equation_Prims.nat", "function_token_typing_Prims.int",
        "haseqTm_refine_542f9d4f129664613f2483a6c88bc7c2",
        "refinement_interpretation_Tm_refine_414d0a9f578ab0048252f8c8f552b99f",
        "refinement_kinding_Tm_refine_542f9d4f129664613f2483a6c88bc7c2"
      ],
      0,
      "1bce0555b365afe8f2c787b333415a91"
    ],
    [
      "TestHasEq.test2",
      1,
      2,
      1,
      [
        "@MaxIFuel_assumption", "@query",
        "Prims_interpretation_Tm_arrow_2eaa01e78f73e9bab5d0955fc1a662da",
        "TestHasEq_interpretation_Tm_arrow_9974df5c311cfcfa7100bc7bef095e1e",
        "assumption_TestHasEq.t__uu___haseq", "equation_Prims.eqtype",
        "equation_Prims.nat", "function_token_typing_Prims.int",
        "haseqTm_refine_2b9cd3e1fbccf82e9ebe34941f00564d",
        "haseqTm_refine_542f9d4f129664613f2483a6c88bc7c2",
        "interpretation_Tm_abs_387fd552e28fe87679131f6c15e8816b",
        "refinement_interpretation_Tm_refine_414d0a9f578ab0048252f8c8f552b99f",
        "refinement_kinding_Tm_refine_542f9d4f129664613f2483a6c88bc7c2",
        "typing_Tm_abs_387fd552e28fe87679131f6c15e8816b"
      ],
      0,
      "5409d6e83f839a7c9a02824e71aec638"
    ],
    [
      "TestHasEq.t1",
      1,
      2,
      1,
      [
        "@MaxIFuel_assumption", "@query", "equation_Prims.eqtype",
        "equation_Prims.nat", "function_token_typing_Prims.int",
        "haseqTm_refine_542f9d4f129664613f2483a6c88bc7c2",
        "refinement_interpretation_Tm_refine_414d0a9f578ab0048252f8c8f552b99f"
      ],
      0,
      "96c1b6b944a8351b27aa22d3477060d1"
    ],
    [
      "TestHasEq.test3",
      1,
      2,
      1,
      [
        "@MaxIFuel_assumption", "@query",
        "assumption_TestHasEq.t1__uu___haseq", "equation_Prims.eqtype",
        "function_token_typing_Prims.bool",
        "refinement_interpretation_Tm_refine_414d0a9f578ab0048252f8c8f552b99f"
      ],
      0,
      "7b8fba42570b9f3540846ae81768a251"
    ],
    [
      "TestHasEq.test4",
      1,
      2,
      1,
      [
        "@MaxIFuel_assumption", "@query",
        "Prims_interpretation_Tm_arrow_2eaa01e78f73e9bab5d0955fc1a662da",
        "TestHasEq_interpretation_Tm_arrow_9974df5c311cfcfa7100bc7bef095e1e",
        "assumption_Prims.dtuple2__uu___haseq", "equation_Prims.eqtype",
        "equation_Prims.nat", "function_token_typing_Prims.int",
        "haseqTm_refine_2b9cd3e1fbccf82e9ebe34941f00564d",
        "haseqTm_refine_542f9d4f129664613f2483a6c88bc7c2",
        "interpretation_Tm_abs_387fd552e28fe87679131f6c15e8816b",
        "refinement_interpretation_Tm_refine_414d0a9f578ab0048252f8c8f552b99f",
        "refinement_kinding_Tm_refine_542f9d4f129664613f2483a6c88bc7c2",
        "typing_Tm_abs_387fd552e28fe87679131f6c15e8816b"
      ],
      0,
      "f1a858150c14ef97587ce7fc58af16d2"
    ],
    [
      "TestHasEq.t4",
      1,
      2,
      1,
      [
        "@MaxIFuel_assumption", "@query", "equation_Prims.eqtype",
        "equation_Prims.nat", "function_token_typing_Prims.int",
        "haseqTm_refine_542f9d4f129664613f2483a6c88bc7c2",
        "refinement_interpretation_Tm_refine_414d0a9f578ab0048252f8c8f552b99f"
      ],
      0,
      "803250b4998090af424149f15a84c53a"
    ]
  ]
]