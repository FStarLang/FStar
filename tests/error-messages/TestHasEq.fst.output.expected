* Info at TestHasEq.fst(58,10-58,11):
  - Expected failure:
  - Failed to prove that the type
    'TestHasEq.t3'
    supports decidable equality because of this argument.
  - Add either the 'noeq' or 'unopteq' qualifier
  - The SMT solver could not prove the query. Use --query_stats for more
    details.
  - See also TestHasEq.fst(57,0-58,19)

* Info at TestHasEq.fst(84,10-84,70):
  - Expected failure:
  - Subtyping check failed
  - Expected type Prims.eqtype got type Type0
  - The SMT solver could not prove the query. Use --query_stats for more
    details.
  - See also TestHasEq.fst(84,12-84,22)

* Info at TestHasEq.fst(88,8-89,30):
  - Expected failure:
  - Invalid qualifiers for declaration `type TestHasEq.erasable_t2`
  - The `unopteq` qualifier is not allowed on erasable inductives since they
    don't have decidable equality.

